### PR TITLE
Allow admins to import volunteers, supervisors, and cases.

### LINKED ISSUE #18: casa_admin import many volunteers
Make a UI where an admin can
1.  import a csv or paste a list of volunteer emails and names to create volunteers in the system. 
2. same for casa cases, with case_number and optional tranistion-aged youth field
3. import list of supervisor-volunteer linkages, supervisor name to volunteer name
4. import list of volunteer-case linkages, volunteer name to case_number

### PR DESCRIPTION
**Pull request by seanmarcia**
### What github issue is this PR for, if any?
Resolves #18 

### What changed, and why?

This PR does a bit more than the issue of just importing volunteers. I also allows the importation of supervisors and casa cases.

I added in the `devise_invitable` gem to make sending emails to new volunteers and supervisors easy. Also added in the mailcatcher gem with some configurations in `development.rb` to enable the testing of sending the emails.

The general process for importing csv's is for folks to download the example csv, edit it, and then upload it.

I also notice that while the dropdown that shows the reports link checks if the person is a casa_admin, the `reports_controller` didn't validate that the user on the page was a casa admin. I updated the controller to use the `before_action :must_be_admin` as well as updated the user in the reports request spec to be a casa_admin.

### How will this affect user permissions?
- Volunteer permissions:
- Supervisor permissions:
- Admin permissions: Only admins can do this.

### How is this tested? (please write tests!) 💖💪
I added in a new imports route that is tested with a request spec and I create a `FileImporter` class that is tested in /spec/models directory. (Probably not the best spot for it.)
I also added in a fixtures directory with example csv's for the three different import types (volunteers, supervisors, casa_cases.)

### Screenshots please :)

menu_item

imports

### CODE REVIEW

**Comment by seanmarcia** [comment]
While I think this is good enough to go right now and begin testing I am planning on refactor to:

Remove the custom routes `import_volunteers`, `import_supervisors`, `import_cases` and be restful and use the create action with a trait to determine which import to run. Probably encapsulate each of the three imports in a service.

I think some of these imports should also be wrapped in an `activeRecord::Base.transaction` -- particularly the casa_case import. Casa cases don't appear to be attached to a casa_org so if we fail to associate them with a user we should probably roll back the transaction otherwise the case may end up floating in the ether. I don't currently have enough context on casa cases (and the rest) to

**Comment by compwron** [comment]
My preferred solution to avoid free-floating cases is, before multi-tenancy, we should add `casa_org_id` to `casa_case`

File: app/views/imports/_supervisors.html.erb [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+    <li><strong>Do not</strong> change any of the values in the first line of the example csv file.</li>
+    <li>Then click the "Import Supervisors CSV" button to import your supervisors.</li>
+
+  </ul>
+  <%= f.file_field :file, accept: 'text/csv', class: 'form-control-file', style: "margin: auto;" %>
+  <br>
+  <%= button_tag :class => "btn btn-md btn-success" do %>
+    <i class="fa fa-upload"></i> Import Supervisors CSV
+  <% end %>
+<% end %>
```
Comment (compwron) [inline app/views/imports/_supervisors.html.erb:21]: Missing trailing linebreaks

File: config/locales/devise_invitable.en.yml [hunk 2]
```diff
@@ -0,0 +1,31 @@
+en:
```
Comment (compwron) [inline config/locales/devise_invitable.en.yml:1]: Yay, locales! :)

File: spec/models/file_importer_spec.rb [hunk 3]
```diff
@@ -0,0 +1,79 @@
+require "rails_helper"
```
Comment (compwron) [inline spec/models/file_importer_spec.rb:1]: Yassssss, tests :)

File: spec/requests/reports_spec.rb [hunk 4]
```diff
@@ -3,7 +3,7 @@
 RSpec.describe "/reports", type: :request do
   describe "GET /index" do
     it "renders a successful response" do
-      sign_in create(:user, :volunteer)
+      sign_in create(:user, :casa_admin)
```
Comment (compwron) [inline spec/requests/reports_spec.rb:6]: 😅 Good fix :)

**Pull request merged**